django/db/backends/base/schema.py

```diff
@@ -629,7 +628,7 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
         if old_db_params['check'] != new_db_params['check'] and old_db_params['check']:
             meta_constraint_names = {constraint.name for constraint in model._meta.constraints}
             constraint_names = self._constraint_names(
-                model, [old_field.column], check=True,
+                model, [old_field.column], check=True, unique=False,
```

Comment:
Check constraints don't seem to be related with this issue, moreover `unique` is `False` for all check constraints. Please revert this unrelated change.

Reply:
@felixxm I explained the reason for this change above. That change was necessary to fix tests failing for MariaDB (there might be some other way but I thought that is the case)

Reply:
Just for the record: Without this in-place `test_alter_field_reloads_state_on_fk_with_to_field_target_type_change (migrations.test_operations.OperationTests)` fails for MariaDB

```
MySQLdb._exceptions.OperationalError: (1005, 'Can\'t create table `test_django`.`alter_alter_field_reloads_state_on_fk_with_to_field_target_t7bb8` (errno: 150 "Foreign key constraint is incorrectly formed")')
```